docs/development/sync-protocol.md

```diff
@@ -0,0 +1,431 @@
+		"parent_name": "launchd",
+		"logged_in_users": [
+			"markowsky"
+		],
+		"quarantine_timestamp": 0,
+		"signing_chain": [{
+				"cn": "Apple Development: Google Development (XXXXXXXXXX)",
+				"valid_until": 1678983513,
+				"org": "Google LLC",
+				"valid_from": 1647447514,
+				"ou": "XXXXXXXXXX",
+				"sha256": "7ae80b9ab38af0c63a9a81765f434d9a7cd8f720eb6037ef303de39d779bc258"
+			},
+			{
+				"cn": "Apple Worldwide Developer Relations Certification Authority",
+				"valid_until": 1897776000,
+				"org": "Apple Inc.",
+				"valid_from": 1582136027,
+				"ou": "G3",
+				"sha256": "dcf21878c77f4198e4b4614f03d696d89c66c66008d4244e1b99161aac91601f"
+			},
+			{
+				"cn": "Apple Root CA",
+				"valid_until": 2054670036,
+				"org": "Apple Inc.",
+				"valid_from": 1146001236,
+				"ou": "Apple Certification Authority",
+				"sha256": "b0b1730ecbc7ff4505142c49f1295e6eda6bcaed7e2c68c5be91b5a11001f024"
+			}
+		],
+		"file_bundle_name": "santasyncservice",
+		"executing_user": "root",
+		"ppid": 1,
+		"file_bundle_path": "/Applications/Santa.app",
+		"file_name": "santasyncservice",
+		"execution_time": 1657764366.475035,
+		"file_sha256": "8621d92262aef379d3cfe9e099f287be5b996a281995b5cc64932f7d62f3dc85",
+		"decision": "ALLOW_BINARY",
+		"file_bundle_id": "com.google.santa.syncservice",
+		"file_bundle_version_string": "9999.1.1",
+		"pid": 2595,
+		"current_sessions": [
+			"markowsky@console",
+			"markowsky@ttys000",
+			"markowsky@ttys001",
+			"markowsky@ttys003"
+		],
+		"team_id": "XXXXXXXXXX"
+	}]
+}
+```
+
+#### `eventupload` Response
+
+The server should reply with an HTTP 200 if the request was successfully received and processed.  
+
+
+| Key | Required | Type | Meaning | Example Value |
+|---|---|---|---|---|
+| event_upload_bundle_binaries | NO | list of strings | An array of bundle hashes that the sync server needs to be uploaded |  |
+
+##### `eventupload` Response Example Payload 
+
+
+```json
+{
+   "event_upload_bundle_binaries": ["8621d92262aef379d3cfe9e099f287be5b996a281995b5cc64932f7d62f3dc85"]
+}
+```
+
+### Rule Download
+
+After Events have been uploaded to the Sync server, the Client then initiates the `rulesdownload` request. 
+
+Like the previous stages this is a simple HTTP request response cycle like so:
+
+```mermaid
+sequenceDiagram
+   client ->> server: POST /ruledownload/<machine_id>
+   server -->> client: ruledownload response
+```
+
+This stage may be performed many times depending on the rule batch size set during the `preflight` stage. 
+
+In the event that the client requested a clean sync or the server responded with a `clean_sync` in the preflight stage then it is expected that the client will throw out its rules and the sync server should supply the client with all of the rules configured for it. 
+
+If a clean sync was not requested by either the client or the sync service, then the sync service should only send new rules seen since the last time the client synced.
```

Comment:
Mention here that rule application is idempotent, so receiving the rules multiple times isn't harmful. This is useful because the safest method for managing this is to update a timestamp when doing a preflight and then in postflight store that timestamp as the new "rule sync" timestamp. Any rules added between that preflight and the ruledownload will be downloaded twice, but that's OK.

Reply:
I took a stab at it.